core/Tracker/TrackerCodeGenerator.php

```diff
@@ -93,6 +115,7 @@ public function generate($idSite, $piwikUrl, $mergeSubdomains = false, $groupPag
 
         $codeImpl = array(
             'idSite'                  => $idSite,
+            // TODO why sanitizeInputValue() and not json_encode?
             'piwikUrl'                => Common::sanitizeInputValue($piwikUrl),
```

Comment:
I'm not sure I understand, if we JSON encode the URL quotes should be escaped so any JS or HTML in there would still be containing in the JS string? I.e. I don't see how there could be XSS.

On the other hand, the way it works now breaks in some (extreme+unrealistic) cases because it encodes HTML entities in a JS context, so it doesn't make sense.